pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -41,11 +41,16 @@ def __init__(self, trainer: "pl.Trainer", resume_from_checkpoint: Optional[_PATH
         self._loaded_checkpoint: Dict[str, Any] = {}
 
     @property
-    def hpc_resume_path(self) -> Optional[str]:
+    def auto_restart_path(self) -> Optional[str]:
         dir_path_hpc = str(self.trainer.weights_save_path)
+        if dir_path_hpc is None or not os.path.isdir(dir_path_hpc):
+            return
```

Comment:
```suggestion
            return None
```
in theory, mypy should complain here

Reply:
Done !